core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -528,6 +528,15 @@ class HttpSamDAO(baseSamServiceURL: String, serviceAccountCreds: Credential)(imp
       callback.future
     }
 
+  override def getUserArbitraryPetServiceAccountKey(userEmail: String): Future[String] =
+    retry(when401or5xx) { () =>
+      val callback = new SamApiCallback[String]("getUserArbitraryPetServiceAccountKey")
+
+      googleApi(rawlsSAContext).getUserArbitraryPetServiceAccountKeyAsync(userEmail, callback)
+
+      callback.future
```

Comment:
Do you know what `callback.future` does here?

Reply:
It's the future that the `SamApiCallback` wraps - so it's just returning that